api/mgo/batch.go

```diff
@@ -0,0 +1,87 @@
+package mgo
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+
+	"github.com/spaceuptech/space-api-go/api/config"
+	"github.com/spaceuptech/space-api-go/api/model"
+	"github.com/spaceuptech/space-api-go/api/proto"
+	"github.com/spaceuptech/space-api-go/api/utils"
+)
+
+// Delete contains the methods for the delete operation
+type Batch struct {
+	ctx    context.Context
+	meta   *proto.Meta
+	db     string
+	config *config.Config
+	reqs   []*proto.AllRequest
+}
+
+type Request interface {
+	getProject()    string
+	getDb()         string
+	getToken()      string
+	getCollection() string
+	getOperation()  string
+}
+
+func initBatch(ctx context.Context, db string, config *config.Config) *Batch {
+	m := &proto.Meta{DbType: db, Project: config.Project, Token: config.Token}
+	return &Batch{ctx, m, db, config, []*proto.AllRequest{}}
+}
+
+// Add adds a delete request to batch
+func (b *Batch) Add(request *Request) (*Batch, error) {
```

Comment:
Let the batch.add() return an error only. It need not return itself.

Reply:
Cool